ggml/src/ggml-cuda/ggml-cuda.cu

```diff
@@ -3744,10 +3744,124 @@ static const char * ggml_backend_cuda_device_get_description(ggml_backend_dev_t
     return ctx->description.c_str();
 }
 
+#if defined(__linux__)
+// Helper function to get available memory from /proc/meminfo for UMA systems
+static bool ggml_backend_cuda_get_available_uma_memory(long * availableMemoryKb, long * freeSwapKb) {
+    FILE * meminfoFile = nullptr;
+    // 2KB buffer for reading /proc/meminfo since it does not report size info, should be enough
+    const size_t BUFFER_SIZE = 2048;
+    auto fileBuffer = std::make_unique<char[]>(BUFFER_SIZE);
+    size_t bytesRead = 0;
+    long hugeTlbTotalPages = -1;
+    long hugeTlbFreePages = -1;
+    long hugeTlbPageSize = -1;
```

Comment:
It would be preferable to use `snake_case`, as it is the standard for this project.